info_encoding.md

```diff
@@ -1,116 +1,108 @@
 # Encoding of the `info` Field in Derivations
 
-The key derivation should include all information of the context
-of the derivation.
-This includes information that is valid only for this particular message,
-as well as the intended use of the key.
-
-Once we have agreed on a suitable selection of values to include, we
-also must encode them in such a way that it can not collide with
-the honest encoding of a different set of values.
-Obviously, an adversary can perform a different encoding, but this
-does not matter, as we assume they can not trick the user into performing
-off-standard derivations or use an off-standard encoding for their derivations.
-Implementations must make sure this assumption holds.
-
-## Notation
-
-In this document, we write lists as comma seperated value, enclosed
-in parentheses. Therefore a list of values 1, 2 and 3 would be written
-as `(1, 2, 3)`.
-Binary concatenation is `||`.
+The key derivation should include all information of the context of the derivation.  This includes information that is valid only for this particular message, as well as the intended use of the key.
+
+Once we have agreed on a suitable selection of values to include, we also must encode them in such a way that it can not collide with the honest encoding of a different set of values.  Obviously, an adversary can perform a different encoding, but this does not matter, as we assume they can not trick the user into performing off-standard derivations or use an off-standard encoding for their derivations.  Implementations must make sure this assumption holds.
+
+## Notation and Definitions
+
+In this document, we write lists as comma seperated value, enclosed in parentheses. Therefore a list of values 1, 2 and 3 would be written as `(1, 2, 3)`.  Binary concatenation is `||`.
+
+For the purpose of this document, common byte order is defined as little endian. In pseudocode, the type `uint16` has the method `encode`, that returns a string containing the common byte order encoding of the number:
+```
+method encode() of uint16 {
+	var out [2]byte
+
+	out[0] = byte(self & 0xff)
+	out[1] = byte(self >> 8)
+
+	return out
+}
+```
+
+For convenience, the function `encodeU16` takes any positive integer as input, converts it to `uint16` and returns the result of `encode`:
+```
+function encodeU16(i int) {
+	assert i >= 0
+	return uint16(i).encode()
+}
+```
```

Comment:
This pseudocode confuses me, because it's too far from languages I'm familiar with. (i.e. I might not be the best to review this).
Are you just defining `UInt16LE` (unsigned integer, 16 bit, little-endian)?
Isn't this fairly common?

Reply:
Also, you say `method ()` and `function ()`

Reply:
I might be peculiar, but I personally prefer a "pictionary" style approach to introducing new ideas.

1. start fuzzy, big picture
  >  "we store things as a concatenated list, so order matters"
2. add more detail
  > "we prefix each thing we concatenate with it's length!"
3. get _real_ specific
  > the encoding of that length is `UInt16LE` , here's exactly what that means, and here's what it all looks like in details when you encode this example string with this example "schema" (the definition of what's in the list and the order)